File: app/src/main/java/com/example/projet/MySQLiteOpenHelper.java
package com.example.projet;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

import androidx.annotation.Nullable;

//room pour
public class MySQLiteOpenHelper extends SQLiteOpenHelper{

    private String creation ="create table Questions("
            +"Parent integer,"
            +"Fils integer primary key,"
            +"Contenu varchar(40) not null ,"
            +"rang integer,"
            +"Username varchar(100) not null);";

    public MySQLiteOpenHelper(@Nullable Context context, @Nullable String name, @Nullable SQLiteDatabase.CursorFactory factory, int version) {
        super(context, name, factory, version);
    }

    @Override
    public void onCreate(SQLiteDatabase db) {
        db.execSQL(creation);
    }

    @Override
    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {

    }
}

File: app/src/main/java/com/keskheu/database/MySQLiteOpenHelper.kt
package com.keskheu.database

import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteDatabase.CursorFactory
import android.database.sqlite.SQLiteOpenHelper

//room pour
class MySQLiteOpenHelper(context: Context?, name: String?, factory: CursorFactory?, version: Int) :
    SQLiteOpenHelper(context, name, factory, version) {
    private val creation = ("create table Questions("
            + "Parent integer,"
            + "Fils integer primary key,"
            + "Contenu varchar(40) not null ,"
            + "rang integer,"
            + "Username varchar(100) not null);")

    override fun onCreate(db: SQLiteDatabase) {
        db.execSQL(creation)
    }

    override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {}
}